poros/python/poros/_compile.py

```diff
@@ -0,0 +1,106 @@
+"""
+compile function for poros.
+"""
+# Copyright (c) 2022 Baidu, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Dict, Any
+import torch
+from torch import nn
+
+import poros._C
+from poros._input_convert import convert_prewarm_inputs
+from poros._input_convert import convert_poros_option
+from poros._module import PorosModule
+
+
+def wrap_cpp_module(cpp_module):
+    """
+    Wrap torch._C.ScriptModule to porosModule, recursively for all submodules
+    """
+    def init_fn(script_module):
+        """init_fn"""
+        for name, cpp_module in torch._C.ModuleDict(script_module._c).items():
+            setattr(script_module, name, wrap_cpp_module(cpp_module))
+        script_module._concrete_type = torch._C.ConcreteModuleType.from_jit_type(script_module._c._type())
+
+        for idx, fn in enumerate(script_module._c._get_forward_pre_hooks()):
+            script_module._forward_pre_hooks[idx] = fn
+        for idx, fn in enumerate(script_module._c._get_forward_hooks()):
+            script_module._forward_hooks[idx] = fn
+
+    return PorosModule._construct(cpp_module, init_fn)
+
+
+def load(filename, poros_options):
+    """
+    todo: load poros model
+    Args:
+        filename( str): poros model save path
+        poros_options(PorosOptions / Dict of settings): compile settings for poros
+    Returns:
+        PorosModule: Compiled Module of poros, 
+                    when run it will partially execute via inlined engine (which is TensorRT)
+    """
+    compiled_cpp_mod = poros._C.load(filename, convert_poros_option(poros_options))
+    compiled_module = wrap_cpp_module(compiled_cpp_mod)
+    return compiled_module
+
+def save(module, filename):
+    """
+    todo: save poros module
```

Comment:
建议这些todo，放在代码注释之后。不然后续开源代码按照注释生成文档时，会带入这些无用信息。
 ```
“”“
Args:
        filename( str): poros model save path
        poros_options(PorosOptions / Dict of settings): compile settings for poros
    Returns:
        PorosModule: Compiled Module of poros, 
                    when run it will partially execute via inlined engine (which is TensorRT)
    """

# TODO: Load poros model
```

按照`# TODO的标准注释`